### PR TITLE
Legend improvements

### PR DESCRIPTION
**Pull request by nf-s**
### Legend improvements

- remove legend broken image icon
- tsxify Legend
- Re-enable geoserver legend scaling
  - Also added geoserver legend font, font size and label margin
  - Add `imageScaling` trait
- Move `LegendTraits` to `CatalogMemberMixin`
- Comment out unused old Legend code
- Extend WMS `isGeoserver` to be true if URL has `geoserver` in it
- Add WMS `supportsGetLegendRequest` trait
- Improved handling of WMS default styles

Fixes https://github.com/TerriaJS/terriajs/issues/4787 and #5021 

A new geoserver legend:
image

### Checklist

-   [x] There are unit tests to verify my changes are correct or unit tests aren't applicable (if so, write quick reason why unit tests don't exist)
-   [x] I've updated CHANGES.md with what I changed.

### CODE REVIEW

**Comment by reginapramesti** [comment]
@nf-s Is there a URL I can try out for this? I could only find https://services.arcgisonline.com/ArcGIS/rest/services/World_Imagery/MapServer/ but it doesn't behave the way it does on your screenshot in the issue 😬

**Comment by nf-s** [comment]
mm that is interesting!
Not sure why that is happening

**Comment by nf-s** [comment]
Sorry @reginapramesti I got distracted!
I might just quickly turn the `Legend` into styled components, so I'll mark this as draft.

## Test URLs:

### WMS with geoserver legend

image

PR: 
- http://ci.terria.io/remove-legend-broken-image-icon/#share=s-QkWhfRsBaPtlo0oeGbtp3qV7gm

Next:
- http://ci.terria.io/next/#share=s-QkWhfRsBaPtlo0oeGbtp3qV7gm

### WMS with non-geoserver legend

PR: 
- http://ci.terria.io/remove-legend-broken-image-icon/#share=s-A0RsiFEvGI4OTSMW8AJvQPD5lgr

Next:
- http://ci.terria.io/next/#share=s-A0RsiFEvGI4OTSMW8AJvQPD5lgr

### WMS with no legend

Use "Delta Blended service" layer

PR: 
- http://ci.terria.io/remove-legend-broken-image-icon/#clean&https://gist.githubusercontent.com/nf-s/3d77d4acb9105283226444677e011dd9/raw/290e89b66e2b2614462b33efb3628dbdd0ac7b99/wms-multidim.json

Next: 
- http://ci.terria.io/next/#clean&https://gist.githubusercontent.com/nf-s/3d77d4acb9105283226444677e011dd9/raw/290e89b66e2b2614462b33efb3628dbdd0ac7b99/wms-multidim.json

### A mix of things to test

http://ci.terria.io/remove-legend-broken-image-icon/#clean&https://gist.githubusercontent.com/nf-s/0077f3c5dd4df0f86b66817a7899496d/raw/ed58e13a1daf3046ec70856d16fda0888d452300/ihatelegends.json

### ODC WMS with default style

https://ows.dea.ga.gov.au/?service=WMS&request=GetCapabilities

### ESRI

http://www.environment.gov.au/mapping/services/ogc_services/CAPAD_MARINE/MapServer/WMSServer

### MapServer

https://www.gebco.net/data_and_products/gebco_web_services/2019/mapserv?service=WMS&version=1.3.0&request=GetCapabilities

**Comment by nf-s** [comment]
Sorry @reginapramesti this has turned into a bit of a monster PR 😶

File: lib/ReactViews/Workbench/Controls/Legend.tsx [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+    if (isImage) {
+      return (
+        <li key={proxiedUrl}>
+          <a
+            href={proxiedUrl}
+            className={Styles.imageAnchor}
+            target="_blank"
+            rel="noreferrer noopener"
+          >
+            <object
```
Comment (KeyboardSounds) [inline lib/ReactViews/Workbench/Controls/Legend.tsx:115]: why are we using `object` here instead of `img` when the object is an image?

**Review by KeyboardSounds — CHANGES_REQUESTED** [review]

**Comment by KeyboardSounds** [comment]
:+1: for adding useful comments to WebMapServiceCatalogItem

File: lib/ReactViews/Workbench/Controls/Legend.tsx (hunk 1 above)
Comment (nf-s) [reply lib/ReactViews/Workbench/Controls/Legend.tsx:115]: This is so we don't get the broken image icon

File: lib/Traits/CatalogMemberTraits.ts [hunk 2]
```diff
@@ -151,6 +151,14 @@ export default class CatalogMemberTraits extends ModelTraits {
   })
   hideLegendInWorkbench: boolean = false;
 
+  @objectArrayTrait({
```
[the remark this answers is not in the document]
Comment (nf-s) [reply lib/Traits/CatalogMemberTraits.ts:154]: There are catalog members outside of `Mappable` which have legends (eg `Chartable`) - I think that if something is able to be added to the Workbench, it should have `LegendTraits` available.
The whole point of me putting `LegendTraits` here is that we had it in random files all over the place.

Also currently you can add groups to the workbench.
We currently do not have a `CatalogItemMixin/Traits` - we may add one in the future if we need to keep groups/items separate

File: lib/Map/Legend.js [hunk 3]
```diff
@@ -1,530 +1,530 @@
-/*global require*/
```
[the remark this answers is not in the document]
Comment (nf-s) [reply lib/Map/Legend.js:1]: yep fair point

File: lib/Traits/LegendTraits.ts [hunk 4]
```diff
@@ -113,6 +113,14 @@ export default class LegendTraits extends ModelTraits {
   })
   url?: string;
 
+  @primitiveTrait({
+    type: "string",
```
[the remark this answers is not in the document]
Comment (nf-s) [reply lib/Traits/LegendTraits.ts]: 👍

**Review by KeyboardSounds — APPROVED** [review]

**Pull request merged**